docker/runner.go

```diff
@@ -22,52 +29,99 @@ func NewRunner() *Runner {
 }
 
 // Run the function at path
-func (n *Runner) Run(f bosonFunc.Function) error {
-	// Check for the docker binary explicitly so that we can return
-	// an extra-friendly error message.
-	_, err := exec.LookPath("docker")
+func (n *Runner) Run(ctx context.Context, f bosonFunc.Function) error {
```

Comment:
IMHO we should add `context.Context` as first parameters to all api functions (e.g. Deploy, Build ...) 
@boson-project/core what do you think?

Reply:
You are probably correct; and they should be added as soon as possible, while the API is still easy to change